uktrade/data-hub-frontend
# Clean Up Components Folder

## Description of change

Following the move of all the components, there's some clean up to do regarding conventions of what and how we export certain components. It's been decided that the following components (ones with Stories) should be the only components to be exported: 

- CompanyLocalHeader
- Badge
- ButtonLink
- CollectionList
- CollectionItem
- CollectionHeader
- CollectionDownload
- CompanyLists
- DeleteCompanyListSection
- Dashboard
- EntityList
- Forms ( check with Ian whether to do forms in this PR )
- FlashMessages
- Main
- Metadata
- NewWindowLink
- Pagination
- Panel
- StatusMessage
- SummaryList
- SummaryTable
- Tag
- Typeahead
- DropdownMenu

These components will be exported from `src/client/components/index.js` in order to have shorter file paths in files that import them. Any components that import other components from the `.../client/components/` folder will have their full paths referenced. Any child/sibling components will not be exported from the main `components` folder, and will have to have their full paths referenced in any files that import them. 
 
After looking at it, moving FlashMessages into it's own folder seems a bit 'out of scope' for this ticket as a couple of the files that use that will need a bit of re writing. Does anyone disagree?

## Test instructions

You should be able to run `npm run storybook` in the front end,  and see nothing missing or broken. The components above should all still be seen in Storybook in your browser. 

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
